src/datasets/utils/file_utils.py

```diff
@@ -568,6 +568,10 @@ def get_from_cache(
         if scheme == "ftp":
             connected = ftp_head(url)
         elif scheme not in ("http", "https"):
+            if scheme in ("s3", "s3a") and storage_options is not None and "hf" in storage_options:
+                # Issue 6598: **storage_options is passed to botocore.session.Session() 
+                # and must not contain keys that become invalid kwargs.
+                del storage_options["hf"]
```

Comment:
it would be great to have a more general solution for any filesystem than hardcoding something for S3

maybe in `_prepare_single_hop_path_and_storage_options()` in file_utils.py ?

Reply:
Thanks ! Can you please explain a bit more what a better  general solution could be?  Is this an option ? Move [this logic](https://github.com/stoicio/datasets/blob/a188022dc43a76a119d90c03832d51d6e4a94d91/src/datasets/download/download_config.py#L96-L97)  to set the hf token to  this protocol check [Link](https://github.com/stoicio/datasets/blob/5e72fb13b4824dcb27aedb807e4e28c420dec244/src/datasets/utils/file_utils.py#L1188)

Reply:
Thanks both, seems the root issue is setting`DownloadConfig.storage_options['hf']` in all situations. We may be able to remove that as `test_load.py` passes without it. It was added at https://github.com/huggingface/datasets/pull/6028 but seems possibly obsoleted by https://github.com/huggingface/datasets/pull/6784.

I've pushed a change. @lhoestq can you please trigger the CI so we can see if this introduces any regressions?

Reply:
sure, I triggered the ci